### DISCUSSION TITLE
Automatically track downloads

### DISCUSSION THREAD

**Discussion by rvlak** [plausible/analytics#1455] [Anything Else]
With Google Analytics and Matomo, file downloads are automatically tracked. It would be great if this could work out-of-the-box with Plausible as well. Especially useful for site owners and newbies who don't know anything about programming with Javascript (like me). I have looked at the [documentation on adding Custom event goals](https://plausible.io/docs/custom-event-goals), but frankly, I don't know how to set this up.

**Update:** You can now track file downloads! All the most common file extensions are supported but you can also specify custom file types to track. Here's how to enable file downloads tracking https://plausible.io/docs/file-downloads-tracking

**Comment by rvlak** [comment]
By the way, if anyone could help me setting this up on our non-profit websites, that would be much appreciated.

**Comment by metmarkosaric** [comment]
thanks for the feedback @rvlak! would be nice if in the future we can make this easier and do something similar for clicks on files as we do for 404 error pages and external link clicks

**Comment by rvlak** [comment]
Good plan, @metmarkosaric! I am convinced that Plausible would be able to get many more new customers on board if download and other event tracking would work out of the box. I know that there is a need for a simple, privacy friendly alternative to Google Analytics, particularly among Wordpress users. Matomo is great in the sense that it's privacy friendly and relatively easy to set up (and measuring downloads and other events works out of the box). But it is only a serious alternative if you are able / can afford to set up your own Matomo server, as it's sluggish and quite instable as a Wordpress plugin. I've had many issues with the Matomo plugin, it has a tendency to suddenly disappear / stop working from time to time, and by the time one finds out about it, there is a gap in the analytics that can never be recovered.

**Comment by metmarkosaric** [comment]
You can now track file downloads! All the most common file extensions are supported but you can also specify custom file types to track. Here's how to enable file downloads tracking https://plausible.io/docs/file-downloads-tracking

**Reply by infosion** [reply]
Thank you! I think i am missing this new feature in the self hosted docker installation, right? My installation ist quite new, about four days so the docker image should be up to date?

**Reply by metmarkosaric** [reply]
Self-hosted is not updated as frequently as the managed hosting version which changes daily. Think of it as a long term release to help us and help you not need to upgrade things all the time and risk breaking stuff. We stack all the new features and fixes that have been battle tested on the hosted version and release a new self-hosted release twice a year. You can learn more here: https://plausible.io/blog/building-open-source

**Reply by infosion** [reply]
Hi Marko, 
thanks for you help - I was just wondering if maybe I had made a mistake myself. 
I find the topic of the long term release excellent - I don't always have to have all features immediately. 
And of course, thanks for plausible in general! It's fun to use and has a clean design. Keep up the great work!
